R/util.R

```diff
@@ -0,0 +1,69 @@
+#' Determine the status of parallel processing. If cores are requested,
+#' set up the cores. If cores are registered, keep using those.
+#' @param parallel logical or integer: whether to use pre-existing parallel
+#'   workers, or how many to create and use. Default `TRUE`.
+#' @param silent logical: give warnings or updates in the computation?
+#'   Default `FALSE`.
+#' @keywords internal
+#' @return number of cores
+setup_parallel <- function(parallel, silent = FALSE) {
+    # Set cluster object to false to start
+    cluster <- FALSE
+    # Check if 'parallel' argument is either logical or numeric
+    if (!is.logical(parallel) && !(parallel %% 1 == 0)) {
```

Comment:
@choisant  This will break if `parallel` is a string like `"1"` (`non-numeric argument to binary operator`). One should first check if `parallel` is numeric (and *not* with `is.integer`, because the value `1` in R is not an integer). Unless of course we're assuming that the user is surely entering a number. Maybe better:
```
if( !(is.logical(parallel) || (is.numeric(parallel) && parallel > 0 && round(parallel) == parallel))) { ...}
```
or similar?

Reply:
Good catch. I'll add that.